### PR TITLE
update titan to include titan#252

### PR DESCRIPTION
**Pull request by waynexia**
Signed-off-by: Ruihang Xia <waynestxia@gmail.com>

update titan submodule to include https://github.com/tikv/titan/commit/7afac725ce134f2112738d4f8a34f0c8a1de863a

### CODE REVIEW

**Comment by tabokie** [comment]
sample_file_size_ratio should be removed from `librocksdb_sys/crocksdb/c.cc` and `librocksdb_sys/crocksdb/crocksdb/c.h`

File: librocksdb_sys/crocksdb/c.cc [hunk 1]
```diff
@@ -6449,10 +6449,6 @@ void ctitandb_options_set_blob_file_discardable_ratio(
   options->rep.blob_file_discardable_ratio = ratio;
 }
 
-void ctitandb_options_set_sample_file_size_ratio(ctitandb_options_t* options,
-                                                 double ratio) {
-  options->rep.sample_file_size_ratio = ratio;
-}
```
Comment (tabokie) [inline librocksdb_sys/crocksdb/c.cc:6452]:
```suggestion
```

File: librocksdb_sys/crocksdb/c.cc [hunk 2]
```diff
@@ -6506,11 +6501,6 @@ void ctitandb_options_set_discardable_ratio(ctitandb_options_t* options,
   options->rep.blob_file_discardable_ratio = ratio;
 }
 
-void ctitandb_options_set_sample_ratio(ctitandb_options_t* options,
```
Comment (ethercflow) [inline librocksdb_sys/crocksdb/c.cc:6509]: It seems that the current TiKV is still using [this](https://github.com/tikv/tikv/blob/master/src/config.rs#L178 ) ? Removing this function will cause a link error in TiKV. @waynexia @tabokie

Comment (ethercflow) [reply librocksdb_sys/crocksdb/c.cc:6509]:
```
  = note: /usr/bin/ld: /workspace/pingcap/tikv/target/x86_64-unknown-linux-gnu/release/deps/tikv_server-d9f913b11da850d2.tikv-6379e790445eb3a2.tikv.4fa2f1af-cgu.0.rcgu.o.rcgu.o: in function `tikv::config::TitanCfConfig::build_opts':
          tikv.4fa2f1af-cgu.0:(.text._ZN4tikv6config13TitanCfConfig10build_opts17h9088003f9f62e419E+0x8b): undefined reference to `ctitandb_options_set_sample_ratio'
          clang: error: linker command failed with exit code 1 (use -v to see invocation)

  = help: some `extern` functions couldn't be found; some native libraries may need to be installed or have their path specified
  = note: use the `-l` flag to specify native libraries to link
  = note: use the `cargo:rustc-link-lib` directive to specify the native libraries to link with Cargo (see https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorustc-link-libkindname)

error: could not compile `tikv-server` due to previous error
warning: build failed, waiting for other jobs to finish...
error: linking with `clang` failed: exit status: 1
```

Comment (tabokie) [reply librocksdb_sys/crocksdb/c.cc:6509]: It's removed in https://github.com/tikv/titan/pull/251. TiKV end needs to deprecate this option.

**Pull request merged**